app/league/slack_messenger.py

```diff
@@ -10,24 +13,23 @@ def __init__(self, app=None):
             self.init_app(app)
 
     def init_app(self, app):
+        self.app = app
         self.url = app.config.get('SLACK_WEBHOOK')
         self.channel = app.config.get('SLACK_CHANNEL')
-        self.payload = {'username': 'webhookbot',
-                        'icon_emoji': ':ghost:',
-                        'channel': self.channel}
+        self.username = app.config.get('SLACK_USERNAME') or DEFAULT_USERNAME
```

Comment:
No need to use `or`, `get` takes a default value as a second param: https://docs.python.org/3.4/library/stdtypes.html#dict.get